ansible_collections/arista/avd/roles/eos_designs/python_modules/inband_management/avdstructuredconfig.py

```diff
@@ -0,0 +1,221 @@
+from __future__ import annotations
+
+from functools import cached_property
+from ipaddress import ip_network
+
+from ansible_collections.arista.avd.plugins.plugin_utils.avdfacts import AvdFacts
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+
+class AvdStructuredConfig(AvdFacts):
+    def __init__(self, hostvars, templar):
+        super().__init__(hostvars, templar)
+
+    @cached_property
+    def _inband_management_role(self) -> str:
+        if (inband_management_role := get(self._hostvars, "switch.inband_management_role")) is not None:
+            return inband_management_role
+
+        if self._inband_management_data:
+            return "parent"
+
+    @cached_property
+    def _inband_management_vlan(self) -> int:
+        return get(self._hostvars, "switch.inband_management_vlan", required=True)
+
+    @cached_property
+    def _inband_management_interface(self) -> str:
+        return get(self._hostvars, "switch.inband_management_interface", required=True)
+
+    @cached_property
+    def _inband_management_ip(self) -> str:
+        return get(self._hostvars, "switch.inband_management_ip", required=True)
+
+    @cached_property
+    def _p2p_uplinks_mtu(self) -> int:
+        return get(self._hostvars, "p2p_uplinks_mtu", required=True)
+
+    @cached_property
+    def _inband_management_gateway(self) -> str:
+        return get(self._hostvars, "switch.inband_management_gateway", required=True)
+
+    @cached_property
+    def _hostname(self) -> str:
+        return get(self._hostvars, "switch.hostname", required=True)
+
+    @cached_property
+    def _mlag(self) -> bool:
+        return get(self._hostvars, "switch.mlag") is True
+
+    @cached_property
+    def _mlag_role(self) -> str | None:
+        return get(self._hostvars, "switch.mlag_role")
+
+    @cached_property
+    def vlans(self) -> dict | None:
+        VLAN_CFG = {
```

Comment:
so do we keep upper case constants inside function or not? (some linters would warn about it though it seems ours does not)?

Reply:
haha PEP say usually, so do I :-p

Reply:
This is not really a constant worthy thing, and I think it would be confusing to move it out. So I have just lower-cased it.